radixdlt-java/src/main/java/com/radixdlt/client/application/translate/unique/AlreadyUsedUniqueIdReasonMapper.java

```diff
@@ -28,4 +34,30 @@
 
 		return Stream.empty();
 	}
+
+	/**
+	 * Extract the affected particle from pointerToIssue of an Atom
+	 *
+	 * @param atom               The atom the particle is in
+	 * @param pointerToIssueJson The pointer to issue in Json form returned by the node
+	 * @return The SpunParticle if it could be extracted
+	 */
+	private Optional<SpunParticle<?>> extractParticleFromPointerToIssue(Atom atom, JsonElement pointerToIssueJson) {
+		try {
+			String pointerToIssue = pointerToIssueJson.getAsString();
+			String groupIndexStr = pointerToIssue.split("/")[2];
+			String particleIndexStr = pointerToIssue.substring(pointerToIssue.lastIndexOf('/') + 1);
+
+			int groupIndex = Integer.parseInt(groupIndexStr);
+			int particleIndex = Integer.parseInt(particleIndexStr);
+
+			ParticleGroup particleGroup = atom.particleGroups().collect(Collectors.toList()).get(groupIndex);
+
+			return Optional.of((SpunParticle<?>) particleGroup.spunParticles().collect(Collectors.toList()).get(particleIndex));
+		} catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
+			LoggerFactory.getLogger(AlreadyUsedUniqueIdReasonMapper.class).error("Malformed pointerToIssue");
```

Comment:
Would prefer to see this done in the usual way, static field with logger initialised at class load.